### ISSUE TITLE
IMAP uses the same mailbox while configured different

### ISSUE THREAD

**Issue by PatchRanger** [mautic/mautic#2405]
| Q  | A
| ---| ---
| Bug report? | y
| Feature request? | n
| Enhancement? | n

## Description:
I've noticed that my email list started to get marked as spam very often and quite quickly.
After some communication with email providers it got clear that incoming FBL-emails are not processed - so users that unsubscribed once still continue to receive my emails. That's why they mark my emails as spam - but still continue to receive them! That's awful.

I've started to investigate the reason.
Running fetch command manually always returns 0 for unsubscribe mailbox. So the issue #1292 looks related.
I've added debug code into the command to see it from inside.
And discovered that the counts of emails fetched by `$imapHelper->searchMailbox('SEEN')` are the same for both mailboxes - they equal the count of bounce mailbox and differ from unsubscribe mailbox.
Looks like Mautic ignores incoming emails for unsubscribe mailbox if it is configured with its own settings.

## Reason
The reason I guess is: Mautic (unintentionally) uses the same IMAP connection for both mailboxes.
So there is no errors anywhere - but the code works wrong.

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.1.0
| PHP version | 5.6.23

### Steps to reproduce:
1. Configure separate mailboxes for bounces and unsubscribes.
2. Get inbox emails for both mailboxes.
3. Run `php app/console mautic:email:fetch --env=prod` manually - or configure to run it on cron and wait until it runs.

Expected: Mails in both mailboxes became seen (read).
Real: Only bounce mailbox got fetched and processed, unsubscribe mailbox stayed untouched - all of the inbox emails are still unread.
 
### Log errors:
No log errors.

**Comment by alanhartless** [comment]
This reads like the issue that was fixed in 2.1.1. https://github.com/mautic/mautic/pull/2303 and https://github.com/mautic/mautic/issues/1472#issuecomment-240508719

**Comment by PatchRanger** [comment]
@alanhartless Oh, I see, thank you, Alan, for quick response.
Usually I prefer to upgrade first before any bug-hunting - but this time I decided to wait until #2390 is fixed.
Also I did a search before posting - but somehow missed the PR you mentioned, sorry for that.

Looks like it is indeed was fixed by aforementioned PR. Let me close this for now - I am going to upgrade and check whether the issue is gone. In case of any regressions related to the issue - I will let you know by re-opening.

**Comment by PatchRanger** [comment]
It's a pity to report: but no, upgrading doesn't help, I am still seeing this situation:
```
3 emails processed for {imap.hosting.example.com:993/imap/ssl}INBOX_bounce@example.com with a criteria of UNSEEN
0 emails processed for {imap.hosting.example.com:993/imap/ssl}INBOX_fbl@example.com with a criteria of UNSEEN
```
While if I enter unsubscribe mailbox manually, I see that there are lots of unseen messages - so I guess "0 emails processed" relates to bounce mailbox. If so, than it means that the issue is still active, so I am re-openeing it.

**Comment by alanhartless** [comment]
What happens if you disable bounce processing and then run the command with just unsubscribe monitoring?

**Comment by PatchRanger** [comment]
@alanhartless I've disabled bounce processing by clearing "Folder to check" setting - as it is written it is intended to completely disable the functionality. Unsubscribe settings leave untouched configured. Then I disabled cron job, which is responsible for fetching. And finally manually run the command of fetch: it works! All of the emails were processed! I've checked it by manually entering the mailbox: all of the emails were marked as read.
